pkg/detectors/fastlypersonaltoken/fastlypersonaltoken.go

```diff
@@ -31,60 +31,34 @@ func (s Scanner) Keywords() []string {
 	return []string{"fastly"}
 }
 
-type fastlyUserRes struct {
-	Login                string `json:"login"`
-	Name                 string `json:"name"`
-	Role                 string `json:"role"`
-	TwoFactorAuthEnabled bool   `json:"two_factor_auth_enabled"`
-	Locked               bool   `json:"locked"`
+type token struct {
+	TokenID   string `json:"id"`
+	UserID    string `json:"user_id"`
+	ExpiresAt string `json:"expires_at"`
+	Scope     string `json:"scope"`
 }
 
 // FromData will find and optionally verify FastlyPersonalToken secrets in a given set of bytes.
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	var uniqueMatches = make(map[string]struct{})
 
-	for _, match := range matches {
-		if len(match) != 2 {
-			continue
-		}
-		resMatch := strings.TrimSpace(match[1])
+	for _, matches := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueMatches[matches[1]] = struct{}{}
+	}
 
+	for match := range uniqueMatches {
 		s1 := detectors.Result{
 			DetectorType: detectorspb.DetectorType_FastlyPersonalToken,
-			Raw:          []byte(resMatch),
+			Raw:          []byte(match),
 		}
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", "https://api.fastly.com/current_user", nil)
-			if err != nil {
-				continue
-			}
-			req.Header.Add("Fastly-Key", resMatch)
-			res, err := client.Do(req)
-			if err == nil {
-				bodyBytes, err := io.ReadAll(res.Body)
-				if err != nil {
-					continue
-				}
-				defer res.Body.Close()
-				if res.StatusCode >= 200 && res.StatusCode < 300 {
-					var userRes fastlyUserRes
-					err = json.Unmarshal(bodyBytes, &userRes)
-					if err != nil {
-						continue
-					}
-					s1.Verified = true
-					s1.ExtraData = map[string]string{
-						"username":                userRes.Login,
-						"name":                    userRes.Name,
-						"role":                    userRes.Role,
-						"locked":                  fmt.Sprintf("%t", userRes.Locked),
-						"two_factor_auth_enabled": fmt.Sprintf("%t", userRes.TwoFactorAuthEnabled),
-					}
-				}
-			}
+			extraData, verified, verificationErr := verifyFastlyApiToken(ctx, match)
+			s1.Verified = verified
+			s1.ExtraData = extraData
+			s1.SetVerificationError(verificationErr)
```

Comment:
I think we're supposed to pass the secret(s) as varargs to SetVerificationErr so they get redacted? Been a while and I'm on my phone so Idk.

I think @ahrav would know.

Reply:
Yeah, it does take secrets as a optional second parameter.